assets/js/common/Table/search.js

```diff
@@ -0,0 +1,23 @@
+const canonicalise = (val) => val.normalize().trim().toLowerCase();
+
+const stringContains = (subject, predicate) =>
+  canonicalise(subject).includes(canonicalise(predicate));
+
+const arrayContains = (subject, predicate) =>
+  subject
+    .map((it) => stringContains(it, predicate))
+    .reduce((accumulator, currentValue) => accumulator || currentValue, false);
+
+const contains = (subject, predicate) =>
+  Array.isArray(subject)
+    ? arrayContains(subject, predicate)
+    : stringContains(subject, predicate);
+
+export const searchByKey = (subject, predicate, ...key) =>
```

Comment:
What about this? (I changed the names to make them easier to understand, but not super set on them)

Note: Also removed the use of the spread operator.

```suggestion
export const search = (row, searchTerm, columnKeys) =>
  columnKeys
    .map((k) => row[k] ? contains(row[k], searchTerm) : false)
    .includes(true);
```